pallets/oracle/src/lib.rs

```diff
@@ -269,15 +269,15 @@ impl<T: Config> Pallet<T> {
 		let converted = match (amount.currency(), currency_id) {
 			(x, y) if x == y => amount.amount(),
 			(_, _) => {
-				// first convert to wrapped, then convert wrapped to the desired currency
-				let base = Self::collateral_to_wrapped(amount.amount(), amount.currency())?; // maybe this does not work?
-				Self::wrapped_to_collateral(base, currency_id)?
+				// first convert to usd, then convert usd to the desired currency
+				let base = Self::usd_to_currency(amount.amount(), amount.currency())?;
+				Self::currency_to_usd(base, currency_id)?
```

Comment:
Done

Reply:
we are not able to switch the function call.
error: test failed, to rerun pass `-p vault --test vault_integration_tests`
so i roll back it.
@ebma

Reply:
CI still failed, so I'm re-running the job now. 
Do you have an idea why changing these two lines made the integration test fail?

Reply:
here is the log when test are not fails:

`is_collateral_below_threshold : true, collateral : XCM(0) 160000000, wrapped_amount : { code: USDC, issuer: GAKNDFRRWA3RPWNLTI3G4EBSD3RGNZZOY5WKWYMQ6CQTG3KIEKPYWAYC } 10000000, threshold : , max_tokens : { code: USDC, issuer: GAKNDFRRWA3RPWNLTI3G4EBSD3RGNZZOY5WKWYMQ6CQTG3KIEKPYWAYC } 0`

here is when it fails
`is_collateral_below_threshold : false, collateral : XCM(0) 1600000, wrapped_amount : { code: USDC, issuer: GAKNDFRRWA3RPWNLTI3G4EBSD3RGNZZOY5WKWYMQ6CQTG3KIEKPYWAYC } 10000000, threshold : FixedU128(1.200000000000000000), max_tokens : { code: USDC, issuer: GAKNDFRRWA3RPWNLTI3G4EBSD3RGNZZOY5WKWYMQ6CQTG3KIEKPYWAYC } 13333333333333333`

the main difference is that `vaultRegistry_getRequiredCollateralForWrapped`
return `1600000`
when it fails instead of `160000000`

Reply:
I also investigated this and the problem is that when registering the vault collateral, the price conversion scales the amount in the wrong way. 
Assuming prices of 1 KSM -> 1 USD and 1 USDC -> 0.1 USD (as it is set up right now in the integration tests):
If we keep the order of `usd_to_currency -> currency_to_usd`, the amount conversion wrongly derives that `16000000` USDC are worth `160000000` KSM.
Switching the order comes to the proper conclusion that `16000000` USDC should be worth `1600000` KSM.